crates/dbsp/src/trace/layers/ordered/mod.rs

```diff
@@ -453,6 +453,108 @@ where
     }
 }
 
+impl<K, L, O> OrderedBuilder<K, L, O>
+where
+    K: Ord + Clone,
+    L: MergeBuilder,
+    O: OrdOffset,
+{
+    /// Like `copy_range`, but uses `fuel` to bound the amount of work.
+    ///
+    /// Builds at most `fuel` values plus values for one extra key.
+    /// If `fuel > 0` when the method returns, this means that the merge is
+    /// complete.
+    pub fn push_merge_fueled(
+        &mut self,
+        input1: (&<Self as Builder>::Trie, &mut usize, usize),
+        input2: (&<Self as Builder>::Trie, &mut usize, usize),
+        fuel: &mut isize,
```

Comment:
We could/should just use a `usize` with `.saturating_sub()`s imo

Reply:
The idea is that we can track over-spent fuel this way.  We don't do this now, but it makes sense for the API to capture this.

Reply:
Interesting, I guess that could be a useful metric